### DISCUSSION TITLE
Schlüsselsymbol im Downloaddialog+ zurückgegebene Bücher immernoch in der Booklib Bibliothek

### DISCUSSION THREAD

**Discussion by alurio** [audiamus/BookLibConnect#14] [Q&A]
Hey Audiamus ! 
Zwei Fragen hätte ich  im Moment 
1Was bedeutet es wenn vor einem Buch ein Schlüssel angezeigt wird ? und verschwindet das irgendwann wieder?
2. Ich habe ein Buch bei Audible zurückgegeben, in der Libary von Booklib wird es aber hartnäckig weiterhin angezeigt. Wie zwinge ich die Software das Buch zu vergessen ? 

Viele Grüße

**Comment by audiamus** [comment]
1. Ist der Schlüssel rot durchgestrichen? Dann bedeutet es, dass kein Schlüssel abgerufen werden konnte. Verschwindet nur, wenn ein neuer Versuch erfolgreich wird. Schlüssel ohne roten Strich: eigentlich ein Zwischenzustand, positive Antwort auf License-Request. Herunterladen des Buches sollte unmittelbar folgen.
2. Ja, da fehlt noch ein Neu-Synchronisieren. Steht auf der Wunschliste. Kommt aber noch nicht sofort. Was wird denn da als Status angezeigt? Herunterladen dürfte nicht mehr gehen. Steht auf der Wunschliste. Kommt aber noch nicht sofort. 

In der Eigenschaftsliste links unten im Bibliotheksfenster steht der aktuelle Status in Textform. Definition:
```
public enum EConversionState {
    unknown,          // question mark
    remote,           // globe
    download,         // globe with down arrow
    license_granted,  // key
    license_denied,   // strikethru key
    downloading,      // down arrow
    download_error,   // strikethru down arrow
    local_locked,     // lock closed
    unlocking,        // key over lock
    unlocking_failed, // strikethru key over lock
    local_unlocked,   // lock open
    exported,         // checkmark mauve
    converting,       // right arrow
    converted,        // checkmark green
    converted_unknown,// checkmark gray
    conversion_error  // red cross
  }
```

**Comment by alurio** [comment]
Hallo !

1. Der Schlüssel ist einfach nur schwarz nicht durchgestrichen.
 2.Das Buch lässt sich noch zum Download auswählen wenn man es dann
versucht runter zu laden gibt es eine unbehandelte Ausnahme irgendwas mit
fehlende Spalten.  Mittlerweile ist das Buch aber weg habe Autoconnect und
Booklib Auto Update einmal deaktiviert, dann nach einem Neustart der
Software wieder aktiviert.  Bisschen Umständlich. Ein neu Syncronisieren
Button wäre sicher nicht schlecht.

Grüße




Am Di., 1. Feb. 2022 um 10:35 Uhr schrieb audiamus ***@***.***

**Reply by audiamus** [reply]
War das dasselbe Buch? Der Herunterladeversuch brach dann ab, ohne weitere Fehlermeldung? Zumindest müsste was im Log stehen. War nach dem nächsten Start von Book Lib Connect der Schlüssel noch da?

Kannst Du mir von der "unbehandelten Ausnahme" die Log-Datei schicken? Gerne auch als Anhang per PM im HiFi-Forum.

**Comment by alurio** [comment]
Hallo ! 
Ne das Buch mit dem Schlüssel ist Terror von Dan simons, das zurückgegebene war der Herr des Wüstenplaneten Was die Logs angeht,  es gibt 10 verschieden logs von Heute keine Ahnung welches die passende ist ich schicke sie dir mal alle  als Zip 
[BookLibConnect Logs.zip](https://github.com/audiamus/BookLibConnect/files/7977634/BookLibConnect.Logs.zip)
Grüße

**Comment by audiamus** [comment]
Danke Dir. "Terror" ist leider nicht dabei. Wohl aber "Wüstenplanet". Damit komme ich schon weiter.

Wenn ich die Zeit finde, gibt es heute Abend erst mal Version 0.3, für die in der letzten Woche gemeldeten Auffälligkeiten.

**Comment by alurio** [comment]
Der Herr des Wüstenplaneten ist jetzt plötzlich doch wieder in der Bibliothek mit rot durchgestrichenem schlüssel.

**Comment by audiamus** [comment]
Hatte mich schon gewundert. Noch gibt es ja keine eingebaute Funktion, Bücher aus der lokalen Datenbank wieder zu entfernen. Wenn man nach der "State"-Spalte sortiert, erscheinen die Bücher nach der o.a. Liste geordnet, also "license denied" zwischen "remote" und "exported". 

Wenn Du noch ein Log für "Terror" finden solltest: ich nehme es gerne. 😀

**Reply by alurio** [reply]
Ich kann eins erzeugen, es steht ja mit schwarzem schlüssel in der Bibliothek

**Reply by audiamus** [reply]
Das wäre nett. Und dann versuchen, das betroffene Buch herunterzuladen.

Falls es schief geht, hätte ich gerne auch die zugehörige `LicenseResponse.json`.

**Comment by alurio** [comment]
Terror ließ sich Problemlos runterladen, der Status in der Bibliothek ist jetzt auf blauer Haken. Brauchst du u nter diesen Umständen trotzdme das Log ?

**Reply by audiamus** [reply]
Danke für den Versuch. Nein, dann dann nützt das Log natürlich nichts. Aber wenn wieder sowas passiert...

Auch wenn nicht abgefangene Ausnahmen auftreten, bin ich sehr an einem Log interessiert - und natürlich an dem Call-Stack, wie bei der ersten Version mit den nicht nummerisch bezeichneten Serientiteln.
